client/container_create.go

```diff
@@ -54,6 +55,19 @@ func (cli *Client) ContainerCreate(ctx context.Context, config *container.Config
 			// When using API under 1.42, the Linux daemon doesn't respect the ConsoleSize
 			hostConfig.ConsoleSize = [2]uint{0, 0}
 		}
+		if versions.LessThan(cli.ClientVersion(), "1.44") {
```

Comment:
Is there a point in doing this if older servers won't be able to return that error (because they don't know about this restriction/or even the new option).

Reply:
Good question; so first of all, this PR is technically just moving code from docker/cli to the client (making the client slightly less error-prone to use).


I think the motivation for adding this validation in docker/cli originally was because in this case the user tries to explicitly opt-in to more restrictive mounts. Failing to do so could lead to the false expectation that the extra restrictions are applied.

(In most other cases, ignoring the option would likely not be too problematic).


But maybe I'm mis-interpreting your comment; let me know!

Reply:
Hmm right, I missed that it's in the `client` not the server-side `api` 🤦🏻

Reply:
Oh! I see the confusion now; thanks!